pdr_backend/lake/persistent_data_store.py

```diff
@@ -196,15 +258,45 @@ def move_table_data(self, temp_table, permanent_table_name: str):
 
         # check if the permanent table exists
         if permanent_table_name not in table_names:
+            # select one row from the temporary table
+            temp_table_df = self.duckdb_conn.execute(
+                f"SELECT * FROM {temp_table.fullname} LIMIT 1"
+            ).pl()
+
+            # create the sql schema
+            sql_table_schema = self._create_sql_table_schema(temp_table_df.schema)
+
+            # create the permanent table
+            self.execute_sql(
+                f"CREATE TABLE {permanent_table_name} ({sql_table_schema})"
+            )
+
             # create table if it does not exist
             self.execute_sql(
-                f"CREATE TABLE {permanent_table_name} AS SELECT * FROM {temp_table.fullname}"
+                f"INSERT INTO {permanent_table_name} SELECT * FROM {temp_table.fullname}"
             )
             return
 
+        # get temporary table columns
+        temp_table_columns = self.duckdb_conn.execute(
+            f"SELECT * FROM {temp_table.fullname} LIMIT 0"
+        ).df()
+
+        temp_table_columns = temp_table_columns.columns
+
+        # generate on conflict clause
+        on_conflict_columns = ", ".join(
+            [
+                f"{column} = EXCLUDED.{column}"
+                for column in temp_table_columns
+                if column != "ID"
+            ]
+        )
```

Comment:
I don't think this is correct...
1. I don't think the pds should handle the conflict. I.e. it should just handle merging from one to the other.
2. You are trying to handle stuff ahead of time.

What should happen right now....
1. Only new events are being processed... **there should be no conflicts.**
2. There will be gaps in the data "null points", do not try to join/fx for these at the moment.